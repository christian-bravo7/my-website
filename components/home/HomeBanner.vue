<template>
  <AppBanner>
    <template #primary-text>
      <span
        class="block | text-center"
      >
        {{ $t('home.iam-name-text') }}
      </span>
      <span
        class="block | text-center | home-banner__description"
        v-html="$md($t('home.iam-job-text'))"
      />
    </template>
    <template #secondary-text>
      {{ $t('home.gretting') }}
    </template>
  </AppBanner>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator';

@Component
export default class HomeBanner extends Vue {}
</script>

<style lang="scss" scoped>
.home-banner {
  &__description {
    /deep/ strong {
      font-weight: normal;

      @extend .gradient-one;
    }
  }
}

html.dark-mode {
  .home-banner {
    &__description {
      /deep/ strong {
        font-weight: normal;

        @extend .gradient-five;
      }
    }
  }
}
</style>
