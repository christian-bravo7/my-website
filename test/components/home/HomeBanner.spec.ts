import { createLocalVue, mount } from '@vue/test-utils';
import MockDate from 'mockdate';

import HomeBanner from '@/components/home/HomeBanner.vue';

import setupI18n from '@/test/config/i18n';

const localVue = createLocalVue();
const i18n = setupI18n(localVue);

const componentConfig = {
  localVue,
  i18n,
};

describe('HomeBanner component', () => {
  it('should render correctly', () => {
    const wrapper = mount(HomeBanner, {
      ...componentConfig,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render primary and seconday text', () => {
    MockDate.set('Wed Nov 25 2020 21:00:00 GMT-0600 (Central Standard Time)');

    const wrapper = mount(HomeBanner, {
      ...componentConfig,
    });

    const primaryText = wrapper.find('.app-banner__primary-text');
    const insideTexts = primaryText.findAll('span.text-center.block');
    const firstText = insideTexts.at(0);
    const secondText = insideTexts.at(1);

    const secondaryText = wrapper.find('#secondary');

    expect(firstText.text()).toBe('I\'m Christian Bravo');
    expect(secondText.text()).toBe('Frontend developer');
    expect(secondaryText.text()).toBe('Good evening');
  });
});
