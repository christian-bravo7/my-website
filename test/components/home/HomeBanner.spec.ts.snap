// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeBanner component should render correctly 1`] = `
<section class="relative | flex justify-center items-center | px-4">
  <div class="relative">
    <h2 id="secondary" class="absolute | font-bold text-center | text-6xl sm:text-8xl md:text-10xl xl:text-13xl | leading-none select-none | text-gray-200 dark:text-blue-900 | app-banner__secondary-text">
      greeting.good-afternoon
    </h2>
    <h1 class="relative z-10 | max-w-xs sm:max-w-lg md:max-w-screen-md | text-center text-xl sm:text-3xl md:text-4xl xl:text-5xl | text-gray-900 dark:text-gray-50 | app-banner__primary-text" style="display: none;"><span class="block | text-center">
      home.iam-name-text
    </span> <span class="block | text-center | home-banner__description">home.iam-job-text</span></h1>
  </div>
  <!---->
</section>
`;
